packages/frontend/src/features/sqlRunner/components/visualizations/Table.tsx

```diff
@@ -0,0 +1,148 @@
+import { type ResultRow } from '@lightdash/common';
+import {
+    flexRender,
+    getCoreRowModel,
+    useReactTable,
+} from '@tanstack/react-table';
+import { useVirtualizer } from '@tanstack/react-virtual';
+import { useRef, type FC } from 'react';
+import { SMALL_TEXT_LENGTH } from '../../../../components/common/LightTable';
+import BodyCell from '../../../../components/common/Table/ScrollableTable/BodyCell';
+import { VirtualizedArea } from '../../../../components/common/Table/ScrollableTable/TableBody';
+import {
+    Table as TableStyled,
+    TableContainer,
+    TableScrollableWrapper,
+    TABLE_HEADER_BG,
+    Tr,
+} from '../../../../components/common/Table/Table.styles';
+import { type useSqlQueryRun } from '../../hooks/useSqlQueryRun';
+import {
+    TableDataTransformer,
+    type TableChartSqlConfig,
+} from '../../transformers/TableDataTransformer';
+
+type Props = {
+    data: NonNullable<ReturnType<typeof useSqlQueryRun>['data']>;
+    config?: TableChartSqlConfig;
+};
+
+export const Table: FC<Props> = ({ data }) => {
+    const processor = new TableDataTransformer(data, undefined); // TODO: add config once we have it
```

Comment:
I'm still a little fuzzy on why we want to use classes here 😅. I think its a pretty uncommon pattern to instantiate classes inside a functional component because kind of the whole point of functional components is that they don't hold onto their own state. Having class instances here skirts the suggested react state management approach where state centrally managed somewhere (redux, a context, etc). 

If we want to do this I think we have to at least put this (as well and the following `get`s) in `useMemo` calls. Looking around the internet I see people saying that's possible, but no one saying its a good idea. I think we are going to have state management problems. 

I think the `TableDataTransformer` class makes sense if we want to use it's functionality on the BE and FE, but the more reacty way to use it here is probably to instantiate it in the parent, then pass it to this component. In some ways that move the problem up the tree, so we can talk about if we want a different pattern. React encourages functional programming pretty heavily and haven't used classes in view components a lot so I don't have good instincts about it.

Reply:
All great points above @magnew, added a class that only takes care of the shared BE-FE methods and created a hook to consume the transformer + intialize React-UI-related code (like `useReactTable` and `useVirtualizer`)